src/distribute.ts

```diff
@@ -0,0 +1,133 @@
+import { writeFileSync } from 'fs';
+import { normalize as normalizePath } from 'path';
+import Transaction from 'arweave/node/lib/transaction';
+import { arweave, readInitialOutputFile } from './common';
+import { communityTxId } from './community/ardrive_contract_oracle';
+import type { Rewards } from './inferno_types';
+
+export type TransactionToDistribute = {
+	id: string;
+	tx: Transaction;
+	target: string;
+	qty: number;
+	rankPosition: number;
+};
+
+// Get the key file
+const keyfile = process.env.KEYFILE ? JSON.parse(process.env.KEYFILE) : undefined;
+
+export async function distributeTokens(confirm: boolean) {
+	const data = readInitialOutputFile();
+	const weeklyRank = data.ranks.weekly;
+
+	if (!keyfile) {
+		throw new Error("There's no keyfile to create transactions");
+	}
+
+	if (!weeklyRank.hasReachedMinimumGroupEffort) {
+		console.log("The group didn't reach the minimum effort this week");
+		return;
+	}
+
+	if (confirm) {
+		console.log('========= CREATING TRANSACTIONS =========');
+	} else {
+		console.log('DRY-RUN: THE SCRIPT IS NOT POSTING TRANSACTIONS');
+	}
+
+	const addresses = weeklyRank.groupEffortRewards.filter((wallet) => wallet.rewards > 0);
+	const transactions = await createTokenDistributionTransactions(addresses);
+
+	if (confirm) {
+		console.log('========= SENDING TRANSACTIONS =========');
+	}
+
+	const transactionsToReport: { id: string; qty: number; rankPosition: number }[] = [];
+
+	for (const transaction of transactions) {
+		if (!confirm) {
+			console.log(transaction);
+			return;
+		}
+
+		const transactionSent = await sendTokenDistributionTransaction(transaction);
+		if (transactionSent) {
+			writeFileSync(normalizePath(`./distribution/${transaction.id}`), JSON.stringify(transaction, null, '\t'));
+			transactionsToReport.push({
+				id: transaction.id,
+				qty: transaction.qty,
+				rankPosition: transaction.rankPosition
+			});
+		}
+	}
+
+	console.log('Writing report to distribution/report.json');
+	writeFileSync(normalizePath('./distribution/report.json'), JSON.stringify(transactionsToReport, null, '\t'));
```

Comment:
Are we gonna use this file in the action? We might have to store it as an artifact, I guess :opionion:

Reply:
We will not use actions to distribute token. We will run manually